### PR TITLE
Helen 80 addition

### PR DESCRIPTION
**Pull request by yj7272098**
## Description

Add Helen 80 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by yj7272098** [comment]
I will go back to this PR when previous PR is done.

**Review by drashna — APPROVED** [review]

File: keyboards/playkbtw/helen80/readme.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+A 80% RGB hotswap keyboard kit made by Play Keyboard.
+
+* Keyboard Maintainer: [Barry Huang](https://github.com/yj7272098)
+* Hardware Supported: Helen 80
+* Hardware Availability: [Play Keyboard](http://play-keyboard.store/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make playkbtw/helen80:default
+
```
Comment (fauxpark) [inline keyboards/playkbtw/helen80/readme.md:14]:
```suggestion

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/playkbtw/helen80/rules.mk [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+LTO_ENABLE = yes
+
+KEY_LOCK_ENABLE = no      # Enable KC_LOCK support
+
+LAYOUTS = ansi_87
```
Comment (fauxpark) [inline keyboards/playkbtw/helen80/rules.mk:27]:
```suggestion
LAYOUTS = tkl_ansi
```

File: keyboards/playkbtw/helen80/info.json [hunk 3]
```diff
@@ -0,0 +1,104 @@
+{
+  "keyboard_name": "helen80",
+  "url": "",
+  "maintainer": "yj7272098",
+  "width": 18.25,
+  "height": 6.5,
+  "layouts": {
+        "LAYOUT_ansi_87": {
```
Comment (fauxpark) [inline keyboards/playkbtw/helen80/info.json:8]:
```suggestion
        "LAYOUT_tkl_ansi": {
```

File: keyboards/playkbtw/helen80/helen80.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_ansi_87( \
```
Comment (fauxpark) [inline keyboards/playkbtw/helen80/helen80.h:21]:
```suggestion
#define LAYOUT_tkl_ansi( \
```

File: keyboards/playkbtw/helen80/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [0] = LAYOUT_ansi_87(
```
Comment (fauxpark) [inline keyboards/playkbtw/helen80/keymaps/default/keymap.c:21]:
```suggestion
  [0] = LAYOUT_tkl_ansi(
```

File: keyboards/playkbtw/helen80/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,       KC_F5,   KC_F6,   KC_F7,   KC_F8,        KC_F9,   KC_F10,  KC_F11,  KC_F12,    KC_PSCR, KC_SLCK, KC_PAUS,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  XXXXXXX, KC_BSPC,   KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,        KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS,       KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
+        KC_LSFT,   XXXXXXX, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,        XXXXXXX,            KC_UP,
+        KC_LCTL,   KC_LGUI,   KC_LALT,                       KC_SPC,                              KC_RALT,   KC_RGUI,   MO(1),     KC_RCTL,      KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+  [1] = LAYOUT_ansi_87(
```
Comment (fauxpark) [inline keyboards/playkbtw/helen80/keymaps/default/keymap.c:30]:
```suggestion
  [1] = LAYOUT_tkl_ansi(
```

File: keyboards/playkbtw/helen80/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [0] = LAYOUT_ansi_87(
```
Comment (fauxpark) [inline keyboards/playkbtw/helen80/keymaps/via/keymap.c:21]:
```suggestion
  [0] = LAYOUT_tkl_ansi(
```

File: keyboards/playkbtw/helen80/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,       KC_F5,   KC_F6,   KC_F7,   KC_F8,        KC_F9,   KC_F10,  KC_F11,  KC_F12,    KC_PSCR, KC_SLCK, KC_PAUS,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  XXXXXXX, KC_BSPC,   KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,        KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS,       KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
+        KC_LSFT,   XXXXXXX, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,        XXXXXXX,            KC_UP,
+        KC_LCTL,   KC_LGUI,   KC_LALT,                       KC_SPC,                              KC_RALT,   KC_RGUI,   MO(1),     KC_RCTL,      KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+  [1] = LAYOUT_ansi_87(
```
Comment (fauxpark) [inline keyboards/playkbtw/helen80/keymaps/via/keymap.c:30]:
```suggestion
  [1] = LAYOUT_tkl_ansi(
```

File: keyboards/playkbtw/helen80/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        _______,          _______, _______, _______, _______,     _______, _______, _______, _______,      _______, _______, _______, _______,   _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,   _______, _______, _______,
+        _______,     RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, _______, _______, _______, _______, _______, _______, _______, _______, _______,        RESET,   _______, _______,
+        _______,       RGB_TOG, RGB_HUD, RGB_SAD, RGB_VAD, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,        _______,            _______,
+        _______,   _______,   _______,                      _______,                              _______,   _______,   _______,   _______,      _______, _______, _______
+    ),
```
Comment (fauxpark) [inline keyboards/playkbtw/helen80/keymaps/via/keymap.c:37]: You need two more empty layers here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by yj7272098** [comment]
@fauxpark Done. Thanks for checking!

**Review by fauxpark — APPROVED** [review]

File: keyboards/playkbtw/helen80/readme.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
```
Comment (Erovia) [inline keyboards/playkbtw/helen80/readme.md:14]: Can you add "clear instructions on how to reset the board into bootloader mode" as per [PR Checklist](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs)?

File: keyboards/playkbtw/helen80/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+  * GNU General Public License for more details.
+  *
+  * You should have received a copy of the GNU General Public License
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  */
+
+#include QMK_KEYBOARD_H
+
+enum _layer {
+    BASE,
```
Comment (Erovia) [inline keyboards/playkbtw/helen80/keymaps/via/keymap.c:20]:
```suggestion
    _BASE,
```

File: keyboards/playkbtw/helen80/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    _RGB,
+    _BLANK1,
+    _BLANK2
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [BASE] = LAYOUT_tkl_ansi(
```
Comment (Erovia) [inline keyboards/playkbtw/helen80/keymaps/via/keymap.c:28]:
```suggestion
  [_BASE] = LAYOUT_tkl_ansi(
```

File: keyboards/playkbtw/helen80/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  * GNU General Public License for more details.
+  *
+  * You should have received a copy of the GNU General Public License
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  */
+
+#include QMK_KEYBOARD_H
+
+enum _layer {
+    BASE,
```
Comment (Erovia) [inline keyboards/playkbtw/helen80/keymaps/default/keymap.c:20]:
```suggestion
    _BASE,
```
I recommend sticking to one format when naming layers.

File: keyboards/playkbtw/helen80/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    _RGB
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [BASE] = LAYOUT_tkl_ansi(
```
Comment (Erovia) [inline keyboards/playkbtw/helen80/keymaps/default/keymap.c:26]:
```suggestion
  [_BASE] = LAYOUT_tkl_ansi(
```

File: keyboards/playkbtw/helen80/keymaps/default/readme.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
+# Default Helen 80 Layout
+
+This is the default layout that comes flashed on every Helen 80. All key pins are shown in the file.
```
Comment (Erovia) [inline keyboards/playkbtw/helen80/keymaps/default/readme.md:3]: > All key pins are shown in the file.

Not sure I know what you mean.

**Review by Erovia — CHANGES_REQUESTED** [review]

**Comment by yj7272098** [comment]
@Erovia Thanks for checking!

**Review by Erovia — APPROVED** [review]

**Pull request merged**